src/app/systelab-components/datepicker/datepicker-time.component.ts

```diff
@@ -29,65 +29,37 @@ export class DatepickerTime extends Datepicker {
 
 	set currentDate(value: Date) {
 		this._currentDate = value;
-		this.currentDateChange.emit(this._currentDate);
 		if (this._currentDate) {
-			this.currentHours = this.currentHours ? this.currentHours : String(this._currentDate.getHours());
-			this.currentMinutes = this.currentMinutes ? this.currentMinutes : String(this._currentDate.getMinutes());
-		} else {
-			this.updateCurrentHours(undefined);
-			this.updateCurrentMinutes(undefined);
-		}
-	}
-
-	@Input()
-	get currentHours(): string {
-		return this._currentHours;
-	}
-
-	set currentHours(value: string) {
-		this.updateCurrentHours(value);
-		if (value && this.currentDate) {
-			if (this._currentHours) {
-				this._currentDate.setHours(Number(value));
+			if (this._currentDate.getHours() === 0 || this.touchSpinHourValues.value !== 0) {
+				this._currentDate.setHours(this.touchSpinHourValues.value);
 			} else {
-				this._currentDate.setHours(0);
+				this.touchSpinHourValues.value = this._currentDate.getHours();
 			}
-			this.currentDateChange.emit(this._currentDate);
-		}
-	}
-
-	@Output() public currentHoursChange = new EventEmitter<string>();
-
-	@Input()
-	get currentMinutes(): string {
-		return this._currentMinutes;
-	}
-
-	set currentMinutes(value: string) {
-		this.updateCurrentMinutes(value);
-
-		if (value && this.currentDate) {
-			if (this._currentMinutes) {
-				this._currentDate.setMinutes(Number(value));
+			if (this._currentDate.getMinutes() === 0 || this.touchSpinMinutesValues.value !== 0) {
+				this._currentDate.setMinutes(this.touchSpinMinutesValues.value);
 			} else {
-				this._currentDate.setMinutes(0);
+				this.touchSpinMinutesValues.value = this._currentDate.getMinutes();
 			}
-			this.currentDateChange.emit(this._currentDate);
+		} else {
+			this.touchSpinHourValues.value = 0;
+			this.touchSpinMinutesValues.value = 0;
+			this.currentHoursChanged(0);
+			this.currentMinutesChanged(0);
 		}
+		this.currentDateChange.emit(this._currentDate);
```

Comment:
I think we have to review if this emit is necessary, because by default will emit in every assignation, even the initialization of a current value saved previously

Reply:
Totally agree and developed.